d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java

```diff
@@ -440,9 +440,11 @@ private static double getErrorRate(Map<ErrorType, Integer> errorTypeCounts, int
     Integer closedChannelExceptionCount = errorTypeCounts.getOrDefault(ErrorType.CLOSED_CHANNEL_EXCEPTION, 0);
     Integer serverErrorCount = errorTypeCounts.getOrDefault(ErrorType.SERVER_ERROR, 0);
     Integer timeoutExceptionCount = errorTypeCounts.getOrDefault(ErrorType.TIMEOUT_EXCEPTION, 0);
+    Integer streamErrorCount = errorTypeCounts.getOrDefault(ErrorType.STREAM_ERROR, 0);
     return callCount == 0
         ? 0
-        : (double) (connectExceptionCount + closedChannelExceptionCount + serverErrorCount + timeoutExceptionCount) / callCount;
+        : (double) (connectExceptionCount + closedChannelExceptionCount + serverErrorCount + timeoutExceptionCount +
+            streamErrorCount) / callCount;
```

Comment:
since this could impact all LB behaviors, we need to be careful and put this change behind a config. cc: @nizarm